js/views/pageNavVw.js

```diff
@@ -591,14 +611,175 @@ module.exports = baseVw.extend({
       Backbone.history.navigate(route, {trigger: true});
     });
   },
+  
```

Comment:
This was my iteration:

1) jQuery plugin (I had some issue with it, can't recall now what it was)
2) a separate view (issues mentioned above)
3) everything in pageNavView

I'll look at it tomorrow again whether it can end up in a plugin. But I think searching through saved data (tags and handles) couldn't be abstract enough because they have a different structure.